flink-runtime/src/test/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinatorTest.java

```diff
@@ -2792,24 +2792,23 @@ public void testSharedStateRegistrationOnRestore() throws Exception {
 
         ExecutionJobVertex jobVertex1 = graph.getJobVertex(jobVertexID1);
 
-        EmbeddedCompletedCheckpointStore store = new EmbeddedCompletedCheckpointStore(10);
-
+        final EmbeddedCompletedCheckpointStore store = new EmbeddedCompletedCheckpointStore(10);
         final List<SharedStateRegistry> createdSharedStateRegistries = new ArrayList<>(2);
 
         // set up the coordinator and validate the initial state
-        CheckpointCoordinator checkpointCoordinator =
+        final CheckpointCoordinatorBuilder coordinatorBuilder =
```

Comment:
The same applies to this test. If we can create the proper checkpoint (incremental ones), then we don't need to create the checkpoint via running the `ExecutionGraph`.

Reply:
I agree with a sentiment of this, but it seems that creating checkpoints via execution graph is actually the easiest approach.

I did some research and it would be probably possible to create a wrapper around PendingCheckpoint (which is hard to setup).

I'd rather leave this one as is for now and maybe create a follow-up issue for refactoring this if you want. This would probably take me another day as it would require deep diving into internal data structures of checkpoints.

Reply:
Ok, this makes sense.